lib/phoenix_component.ex

```diff
@@ -1,290 +1,336 @@
-
-  Here is an example:
+  A function component is any function that receives an assigns map as an argument and returns 
+  a rendered struct built with [the `~H` sigil](`Phoenix.LiveView.Helpers.sigil_H/2`):
 
       defmodule MyComponent do
         use Phoenix.Component
-
-        # Optionally also bring the HTML helpers
-        # import Phoenix.LiveView.Helpers
-
+    
         def greet(assigns) do
           ~H"""
-          <p>Hello, <%= assigns.name %></p>
+          <p>Hello, <%= @name %>!</p>
           """
         end
       end
 
-  The component can be invoked as a regular function:
+  When invoked within a `~H` sigil or HEEx template file:
 
-      MyComponent.greet(%{name: "Jane"})
+      <MyComponent.greet name="Jane" />
 
-  But it is typically invoked using the function component
-  syntax from the `~H` sigil:
+  The following HTML is rendered:
 
-      ~H"""
-      <MyComponent.greet name="Jane" />
-      """
+      <p>Hello, Jane!</p>
 
-  If the `MyComponent` module is imported or if the function
-  is defined locally, you can skip the module name:
+  If the function component is defined locally, or its module is imported, then the caller can
+  invoke the function directly without specifying the module:
 
-      ~H"""
       <.greet name="Jane" />
-      """
 
-  Similar to any HTML tag inside the `~H` sigil, you can
-  interpolate attributes values too:
+  For dynamic values, you can interpolate Elixir expressions into a function component:
 
-      ~H"""
       <.greet name={@user.name} />
-      """
 
-  You can learn more about the `~H` sigil [in its documentation](`Phoenix.LiveView.Helpers.sigil_H/2`).
+  Function components can also accept blocks of HEEx content (more on this later):
+
+      <.card>
+        <p>This is the body of my card!</p>
+      </.card>
 
-  ## `use Phoenix.Component`
+  Like `Phoenix.LiveView` and `Phoenix.LiveComponent`, function components are implemented using
+  a map of assigns, and follow [the same rules and best practices](../guides/server/assigns-eex.md).
+  However, we typically do not implement function components by manipulating the assigns map 
+  directly, as `Phoenix.Component` provides two higher-level abstractions for us: 
+  attributes and slots.
 
-  Modules that define function components should call
-  `use Phoenix.Component` at the top. Doing so will import
-  the functions from both `Phoenix.LiveView` and
-  `Phoenix.LiveView.Helpers` modules. `Phoenix.LiveView`
-  and `Phoenix.LiveComponent` automatically invoke
-  `use Phoenix.Component` for you.
+  Note that attributes and slots requires the `:phoenix_live_view` compiler to be added to your
+  `:compiler` options in your `mix.exs`'s `project` configuration:
 
-  You must avoid defining a module for each component. Instead,
-  we should use modules to group side-by-side related function
-  components.
+      def project do
+        [
+          # ...
+          compilers: [:phoenix_live_view] ++ Mix.compilers(),
```

Comment:
Note for the future, the compiler is going away as soon as I can dedicate some time to remove it